packages/client/src/Stream.ts

```diff
@@ -203,25 +203,27 @@ class StreamrStream implements StreamMetadata {
         return this._streamRegistry.hasPublicPermission(this.id, permission)
     }
 
-    // async hasUserPermissions(operations: StreamPermission[], userId: string) {
-    //     this.assertUserId(userId)
-    //     const permissions = this._streamRegistry.getPermissionsForUser(this.id, userId, )
-    //     return this.hasPermissions(operations, userId)
-    // }
-    // async hasPermissions(operations: StreamPermission[], userId: string) {
-    //     const permissions = await this._streamRegistry.getPermissionsForUser(this.id, userId)
-
-    //     if (operation === StreamPermission.PUBLISH || operation === StreamPermission.SUBSCRIBE) {
-    //         return permissions[operation].gt(Date.now())
-    //     }
-    //     return permissions[operation]
-    // }
-
-    // async hasPermission(operation: StreamPermission, userId: string|undefined) {
-    //     const permissions = await this.hasPermissions([operation], userId)
-    //     if (!Array.isArray(permissions) || !permissions.length) { return undefined }
-    //     return permissions[0]
-    // }
+    async hasUserPermissions(requestedPermissions: StreamPermission[], userId: string): Promise<boolean> {
+        this.assertUserId(userId)
+        const permissions = await this._streamRegistry.getDirectPermissionsForUser(this.id, userId)
```

Comment:
Maybe could condense the `for..of` to:
```typescript
return requestedPermission.every((requestedPermission) => permissions[requestedPermission])
```
Though perhaps there's an argument to keeping it like this for some kind of consistency with the other methods below?